### PR TITLE
feat: Generic VC Issuer

### PR DESCRIPTION
**Pull request by kdimak**
Signed-off-by: Dmitriy Kinoshenko <dkinoshenko@gmail.com>

#1767

### CODE REVIEW

File: pkg/doc/verifiable/credential.go [hunk 1]
```diff
@@ -391,9 +391,58 @@ type Evidence interface{}
 
 // Issuer of the Verifiable Credential
 type Issuer struct {
-	ID    string
-	Name  string
-	Image string
+	ID string `json:"id,omitempty"`
+
+	CustomFields CustomFields `json:"-"`
```
Comment (troyronda) [inline pkg/doc/verifiable/credential.go:396]: note for future field - i think the spec usually refers to this as extensions

File: pkg/doc/verifiable/credential_test.go [hunk 2]
```diff
@@ -78,8 +78,8 @@ func TestNewCredential(t *testing.T) {
 		// validate not null credential subject
 		require.NotNil(t, vc.Issuer)
 		require.Equal(t, "did:example:76e12ec712ebc6f1c221ebfeb1f", vc.Issuer.ID)
-		require.Equal(t, "Example University", vc.Issuer.Name)
-		require.Equal(t, "data:image/png;base64,iVBOR", vc.Issuer.Image)
+		require.Equal(t, "Example University", vc.Issuer.CustomFields["name"])
+		require.Equal(t, "data:image/png;base64,iVBOR", vc.Issuer.CustomFields["image"])
```
Comment (troyronda) [inline pkg/doc/verifiable/credential_test.go:82]: do we have a test somewhere to ensure that custom fields are marshalled when outputting JSON?

**Comment by troyronda** [comment]
@kdimak please rebase.

File: pkg/doc/verifiable/credential_test.go (hunk 2 above)
Comment (kdimak) [reply pkg/doc/verifiable/credential_test.go:82]: Yes, I added a new test called `TestMarshalIssuer()` for this purpose.

**Comment by kdimak** [comment]
> @kdimak please rebase.

done

**Pull request merged**